unicodetools/src/main/java/org/unicode/text/UCD/GenerateData.java

```diff
@@ -811,6 +814,57 @@ public static void writeNormalizerTestSuite(String directory, String fileName)
         }
+        // At least one implementation (ICU4X) has an edge case when a character
+        // whose decomposition contains multiple starters and ends with a
+        // non-starter is followed by a non-starter of lower CCC.
+        // See https://github.com/unicode-org/unicodetools/issues/656
+        // and https://github.com/unicode-org/icu4x/pull/4530.
+        // That implementation also has separate code paths for the BMP and
+        // higher planes.  No such decompositions currently exist outside the
+        // BMP, but by generating these test cases we ensure that this would be
+        // covered.
+        // We stick them in Part 0, which is in principle for handcrafted test
+        // cases, because there are not many of them, and the edge case feels a
+        // tad too weird to describe in the title of a new part.
+        final org.unicode.props.UnicodeProperty sc =
+                IndexUnicodeProperties.make().getProperty(UcdProperty.Script);
+        final org.unicode.props.UnicodeProperty ccc =
+                IndexUnicodeProperties.make().getProperty(UcdProperty.Canonical_Combining_Class);
+        loopOverCodePoints:
+        for (final String cp : UnicodeSet.ALL_CODE_POINTS) {
+            final String[] decompositions =
+                    new String[] {Default.nfd().normalize(cp), Default.nfkd().normalize(cp)};
+            for (final String decomposition : decompositions) {
+                final int lastCCC =
+                        Default.ucd()
+                                .getCombiningClass(
+                                        decomposition.codePointBefore(decomposition.length()));
+                final long nonStarterCount =
+                        decomposition
+                                .codePoints()
+                                .filter(c -> (Default.ucd().getCombiningClass(c) == 0))
+                                .count();
+                final String script = sc.getValue(cp.codePointAt(0));
+                if (lastCCC > 1 && nonStarterCount > 1) {
+                    // Try to pick a trailing nonstarter that might have a
+                    // chance of combining with the character if possible,
+                    // both for æsthetic reasons and to reproduce the example
+                    // ICU4X came across.  If all else fails, use a character
+                    // with CCC=1, as low as it gets.
+                    if (script.equals("Arabic") && lastCCC > 220) {
+                        // ARABIC SUBSCRIPT ALEF.
+                        writeLine(cp + "\u0656", log, false);
+                    } else if (lastCCC > 220) {
+                        // COMBINING DOT BELOW.
+                        writeLine(cp + "\u0323", log, false);
+                    } else {
+                        // COMBINING TILDE OVERLAY.
+                        writeLine(cp + "\u0334", log, false);
+                    }
+                    continue loopOverCodePoints;
```

Comment:
Nit: A simple `break` would suffice here, and eliminate the outer-loop label.

Reply:
Done.